Hw21 aot

joncham opened the pull request:
<!--
Thank you for your Pull Request!

If you are new to contributing to Mono, please try to do your best at conforming to our coding guidelines http://www.mono-project.com/community/contributing/coding-guidelines/ but don't worry if you get something wrong. One of the project members will help you to get things landed.

Does your pull request fix any of the existing issues? Please use the following format: Fixes #issue-number
-->

lucasmeijer committed:
enable llvm

lucasmeijer committed:
first stab at some CI

AndreasReich committed:
bcl aot wip

lucasmeijer committed:
fix silly

lucasmeijer committed:
set MONO_PATH

lucasmeijer committed:
make unity buildscript ship llc.exe

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/mono?pullRequest=1452) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/mono?pullRequest=1452) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: AndreasReich<br/>:x: amirebrahimi<br/>:x: ecuzzillo<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/mono?pullRequest=1452) it.</sub>

joncham committed:
Apply aot domain support from branch unity-aot-domain.

AndreasReich committed:
fix win64 build not bundling llvm binaries

AndreasReich committed:
[skip-ci] don't use llvm enable on win32

AndreasReich committed:
fix libpaths for aot

AndreasReich committed:
fix using wrong mono.exe for aot

joncham committed:
WIP move AOT to domain data structures.

Right now AOT data is stored on the image and "leaks" into the root
domain under the assumption that it will always be run there. By moving
the MonoAotModule to the MonoDomain we can better control and track
the lifetime of AOT data.

AndreasReich committed:
enable mscorlib aot

joncham committed:
Force write barriers in Boehm.

This ensures the AOT compile produces write barriers which Unity may
enable.

lucasmeijer committed:
distribute .pdb files

joncham committed:
Use AOT domain when looking up AOT unwind info.

ecuzzillo committed:
update llvm commit to elliot fork to add burst pass

ecuzzillo committed:
Merge branch 'hw21-aot' of github.com:Unity-Technologies/mono into hw21-aot

ecuzzillo committed:
forgot to update .gitmodules

amirebrahimi committed:
Adding some heuristics to skip LLVM use in certain extremely bad situations

joncham committed:
Lookup cached class in aot domain.

joncham committed:
Free memory allocated for AOT module.
